ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -299,35 +304,119 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) {
     };
   }
 
+  private void startTransaction(StreamInfo info, DataStreamRequestByteBuf request, ChannelHandlerContext ctx) {
+    try {
+      server.submitClientRequestAsync(info.getRequest()).thenAcceptAsync(reply -> {
+        if (reply.isSuccess()) {
+          ByteBuffer buffer = ClientProtoUtils.toRaftClientReplyProto(reply).toByteString().asReadOnlyByteBuffer();
+          sendReplySuccess(request, buffer, -1, ctx);
+        } else {
+          if (request.getType() == Type.STREAM_CLOSE) {
+            // if this server is not the leader, forward start transition to the other peers
+            forwardStartTransaction(info, request, ctx);
```

Comment:
I might have missed something: so fail to apply `STREAM_CLOSE` to RaftServer can conclude to that the server is not the leader?

Reply:
@amaliujia Could not conclude that, not the leader is only one case, there maybe other unexpected cause. But we should forwardStartTransaction anyway.

Reply:
Got it. I was bit confused on the comment here :)

Reply:
Let me update the comment